src/main/groovy/org/shipkit/internal/gradle/plugin/PluginValidator.java

```diff
@@ -0,0 +1,93 @@
+package org.shipkit.internal.gradle.plugin;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+
+import java.io.File;
+import java.util.*;
+
+import static org.shipkit.internal.gradle.plugin.PluginUtil.DOT_PROPERTIES;
+
+
+public class PluginValidator {
+
+    private static final Logger LOG = Logging.getLogger(PluginValidator.class);
+
+    public void validate(Set<File> gradlePlugins, Set<File> gradleProperties) {
+        ensurePluginsHavePropertiesFile(gradlePlugins, gradleProperties);
+    }
+
+    private void ensurePluginsHavePropertiesFile(Set<File> gradlePlugins, Set<File> gradleProperties) {
+        Map<String, File> missingPropertiesFiles = new HashMap<String, File>();
+        for (File plugin : gradlePlugins) {
+            String pluginFilename = plugin.getName();
+            String pluginName = extractPluginName(pluginFilename);
+            if (pluginName != null) {
+                String convertedPropertiesFleName = convertToPropertiesFileName(pluginName);
+                boolean matchingPropertiesFound = false;
+                for (File properties : gradleProperties) {
+                    int lastIndexOfProperties = properties.getName().lastIndexOf(DOT_PROPERTIES);
+                    if (lastIndexOfProperties != -1) {
+                        String pluginExtractedFromProperties = properties.getName().substring(0, lastIndexOfProperties);
+                        if (pluginExtractedFromProperties.toLowerCase().endsWith(convertedPropertiesFleName.toLowerCase())) {
+                            matchingPropertiesFound = true;
+                            LOG.info("plugin " + plugin + " has properties file " + gradleProperties);
+                            break;
+                        }
+                    }
+                }
+                if (!matchingPropertiesFound) {
+                    missingPropertiesFiles.put(pluginName, plugin);
+                }
+            }
+        }
+
+        throwExceptionIfNeeded(missingPropertiesFiles);
+    }
+
+    private void throwExceptionIfNeeded(Map<String, File> missingPropertiesFiles) {
+        if (missingPropertiesFiles.size() > 0) {
+            StringBuilder sb = new StringBuilder();
+            sb.append("no properties file found for plugin(s):");
+            for (String pluginName : missingPropertiesFiles.keySet()) {
+                sb.append("\n\t");
+                sb.append("'");
+                sb.append(pluginName);
+                sb.append("' (");
+                sb.append(missingPropertiesFiles.get(pluginName));
+                sb.append(")");
```

Comment:
Looks nice. What do you think about:
```
for (...) {
   sb.append(String.format("LINE_SUFFIX'%s' (%s)", pluginName, missingPropertiesFile.get(pluginName));
}
```
Number of plugins is limited and performance should not be a problem.